### ISSUE TITLE
Web interface shredder config.pl

### ISSUE THREAD

**Issue by bitranox** [backuppc/backuppc#499]
if You change the configuration in Web Interface : 
Backupplan / FullKeeCnt to an array like 4, 2, 2 and press "save"
BackupPc hangs up, and /etc/config.pl is invalid.

**Comment by Caprico85** [comment]
Duplicate of https://github.com/backuppc/backuppc/issues/466 and #472, fixed by https://github.com/backuppc/backuppc/commit/2c9270b9b849b2c86ae6301dd722c97757bc9256.

Unfortunately no new release yet (and I wonder if there will ever be one as BackupPC seems mostly dead now).

**Comment by bitranox** [comment]
very cool, thanks, it works.
if BackupPC is mostly dead - any suggestions for a good replacement ?

**Comment by Caprico85** [comment]
> if BackupPC is mostly dead - any suggestions for a good replacement ?

No, unfortunately not. I'm looking also for a good replacement. I looked at several options, but discarded them all:

- UrBackup: Weird installation, where you download some shell scripts from all over the web to install it. I am not going to run a shell script from some unknown guy on my production servers. No pre-built packages in your distribution's package repository. Seems way too hacky for me.

- Borg: No UI, no schedules. More like rsync on speed. You run it, it copies the files to a target system and thats it. Enough for backing up your home PC to an external drive, but nothing for enterprise.

- Bacula, Veeam: Cost money (my boss won't pay)

- Bareos: Looked interesting at first. Quite difficult to understand. Totally geared towards tape libraries. Wastes too much space. Each full backup takes up the full space again (where BackupPC deduplicates between full backups). Backups are stored in "volume files" (which are basically tape images), with apparently no way to clean up free space on old tape files. If 100 bytes on an old 50 GB tape file are still used, it'll keep the whole 50 GB file. Maybe all of this can be fixed with some obscure config settings, but I'm not spending any more time on it.

So, for now I'm staying with BackupPC. I may move away from adferrand's docker image to an own image where I can hack and patch around as I like.

**Comment by SpareSimian** [comment]
What does the config file look like before and after the save? What actually hangs up? The web interface or the daemon? 

I've been using BackupPC on CentOS 7 and will likely have to move it to a new OS, perhaps Alma 9 or Debian Bookworm, when CentOS 7 goes EOL in a couple of months. It does what I need, so I'm hoping none of its dependencies will cause problems when updated.

**Comment by fauust** [comment]
See: https://github.com/backuppc/backuppc/issues/472

@Caprico85 
Other alternative worth mentioning, restic and borgmatic. None of them comes with all of the BackupPC features though, it's really sad to see this great piece of software abandoned :(

**Comment by SpareSimian** [comment]
@fauust as evidenced by #472, that's fixed. Does your installation lack that fix?

**Comment by fauust** [comment]
@SpareSimian, fixed but not released, I replaced BackupPC with Restic for that reason, no release since June 2020.

**Comment by GWHAYWOOD** [comment]
Closed - fixed in master branch.
